applications/P2PStateChannels.md

```diff
@@ -0,0 +1,254 @@
+
+Peer-to-peer (p2p) state channels have many benefits—they’re highly scalable and decentralized due to their network topology and tendency to partition state within their group of interaction. They offer real-time communication that’s ‘free’ (no transaction fee) but do not act like a trusted party like blockchains do.
+
+What we aim to achieve is combine the best of both worlds, the decentralized and scalable nature of full p2p systems, while also inheriting full security from a distributed ledger. 
+
+Notice that society works in the exact same way. For most of our lives, we (peers) interact with other peers in small groups (partitions). We (peers) form agreements and commit to faithfully executing on them and in the case where those agreements fail, we go to a court system to dispute one another inorder to enforce those agreements. The courts have limited capacity and their intervention is slow and expensive, but they are trusted by all parties (that freely chose to be in that jurisdiction) to be impartial, objective and faithfully come to a verdict on the agreement/law (state machine).
+
+### How does this relate to Polkadot
+The value proposition for Polkadot is improving the user experience by providing a ‘free’ real-time execution environment which all parachains currently lack and expanding the market by onboarding new users and developers into the ecosystem.
+
+# Project Details
+
+This section will be used to expand on the technical solution and the architecture. 
+
+The system is composed of a client side framework/SDK and on-chain smart contracts.
+
+Tech stack:
+On-chain: smart contracts (solidity)
+Off-chain (p2p) - written in typescript, holepunch for p2p and discovery, ethereumjs for running the EVM on client, ethers for manipulating and signing evm data and everything else is built from scratch (typescript). The entire design is state machine and runtime independent (can support substrate) - it works with native tooling. 
+
+### The main responsibilities of the client side framework are:
+1) Networking - establishing direct peer-to-peer connections and discovery over various transports (currently Holepunch),
+2) State machine replication - allowing connected peers to commit to state machines and allowing them to progress and replicate those state machines,
+3) Agreement tracking - tracking agreements on the order of transactions and state, managing forks, checking and collecting signatures, finalizing state
+4) Virtual (logical) clock - Having a local logical clock that’s synced to the DLT clock (timestamp) - allows peers to have a local perception of time that mimics the one of the DLT, but tolerates skewness to a degree. This is of course subjective and peers will never reach consensus on the clock nor with other peers nor with the DLT itself. This is very important and will be discussed further below,
+5) Dispute handling - constructing fraud proofs when a peer deviates from the protocol,
```

Comment:
I guess "fraud proof" is fine here, but mostly it just returns to running on-chain

The envisioned applciations have timeout information which after that concludes the execution can theoretically move back off-chain, although maybe that's an optimization not worth the development costs. I guess nothing much to change here.

Reply:
Yes